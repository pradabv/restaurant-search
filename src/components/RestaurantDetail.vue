<template>
    <div>
        <div v-if="selectedItem">
            <div class="restaurant-details">
                <div class="retaurant-header">
                    <h2><a :href="selectedItem.url" :target="'_blank'">{{selectedItem.name}}</a></h2>
                    <p><b>Rating:</b> {{selectedItem.rating}}</p>
                    <h4 class="location">
                        <img src="../assets/icons/marker.svg" alt="restaurant location" title="restaurant location">
                        <span>{{selectedItem.location.formatted_address}}</span>
                    </h4>
                    <h4 class="location" v-if="selectedItem.display_phone">
                        <img src="../assets/icons/phone.svg" alt="restaurant number" title="restaurant number">
                        <span>{{selectedItem.display_phone}}</span>
                    </h4>
                </div>
                <div class="retaurant-img" v-for="(img, index) of selectedItem.photos" :key="index"> 
                    <img :src="img" :alt="selectedItem.name" :title="selectedItem.name">
                </div>
            </div>
            <div class="restaurant-reviews">
                <h5>Reviews</h5>
                <ul v-if="selectedItem.reviews.length">
                    <li v-for="(review, index) in selectedItem.reviews" :key="index"> 
                        <p><b>Rating:</b> {{review.rating}}</p>
                        <p>{{review.text}}</p>
                    </li>
                </ul>
                <p v-else>No reviews found</p>
            </div>
        </div>
        <div v-else class="welcome-screen">
            <h3>Welcome to Restaurant Finder</h3>
            <p>Please Select a Restaurant to view details</p>
        </div>
    </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';
import store from '@/store';

@Component
export default class RestaurantDetail extends Vue {
  mounted() {
    store.dispatch('getJsonData');
  }

  get selectedItem() {
    return store.state.selectedItem;
  }

}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="scss">

.restaurant-details{
    flex-direction: row;
    box-sizing: border-box;
    display: flex;
    place-content: flex-start;
    align-items: flex-start;
    padding: 40px;
}
.welcome-screen{
   padding: 15px 250px;
}
.retaurant-header{
    width: 70%;
}
.retaurant-header h2{
    text-decoration: underline;
    margin-bottom: 20px;
}
.retaurant-header a {
    color:  #282c34;
}
.retaurant-header h4.location{
    font-size: 16px;
    margin: 20px 0px;
    font-weight: normal;
}

.retaurant-img{
    width: 30%;
}
.retaurant-img img{
    width: 100%;
    height: auto;
}

.restaurant-reviews{
    padding: 0px 40px;
}
.restaurant-reviews h5{
    font-size: 20px;
}
.restaurant-reviews ul{
    height: 30vh;
    padding-right: 30px;
    overflow-y: auto;
    overflow-x: hidden;
}
</style>


