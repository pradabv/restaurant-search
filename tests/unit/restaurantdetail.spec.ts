import { shallowMount, createLocalVue } from '@vue/test-utils';
import RestaurantDetail from '@/components/RestaurantDetail.vue';
import Vuex from 'vuex';
import {sampleData} from '../mockData/mock.data';
import Store from './../../src/store/index';

const localVueWithVuex = createLocalVue();
localVueWithVuex.use(Vuex);

describe('RestaurantDetail.vue', () => {
  let store: any;
  
  beforeEach(() => {
    store = new Vuex.Store(Store);
  });

  it('RestaurantDetail has welcome screen before set data in store', () => {
    const wrapper = shallowMount(RestaurantDetail, {
      localVue: localVueWithVuex,
      store
    });
    expect(wrapper.findAll('.welcome-screen')).toBeTruthy();
  });

  it('RestaurantDetail has welcome screen before set data in store', () => {
    Store.commit('setSelectedItem', sampleData[0]);
    const wrapper = shallowMount(RestaurantDetail, {
      localVue: localVueWithVuex,
      store
    });
    expect(wrapper.findAll('.restaurant-details')).toBeTruthy();
  });


  it('RestaurantDetail home screen MatchSnapshot', () => {
    const wrapper = shallowMount(RestaurantDetail);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('RestaurantDetail MatchSnapshot after select item', () => {
    Store.commit('setSelectedItem', sampleData[0]);
    const wrapper = shallowMount(RestaurantDetail);
    expect(wrapper.html()).toMatchSnapshot();
  });

  });
  