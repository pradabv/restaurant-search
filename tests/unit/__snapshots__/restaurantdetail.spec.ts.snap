// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantDetail.vue RestaurantDetail MatchSnapshot after select item 1`] = `
<div>
  <div>
    <div class="restaurant-details">
      <div class="retaurant-header">
        <h2><a href="https://www.yelp.com/biz/billy-the-butcher-hamburg-2?adjust_creative=nMAD8ywU1QBl_RMSM215Bg&amp;utm_campaign=yelp_api_v3&amp;utm_medium=api_v3_graphql&amp;utm_source=nMAD8ywU1QBl_RMSM215Bg" target="_blank">Billy The Butcher</a></h2>
        <p><b>Rating:</b> 4</p>
        <h4 class="location"><img src="../assets/icons/marker.svg" alt="restaurant location" title="restaurant location"> <span>Jungfernstieg 16 - 20
4. OG
20354 Hamburg
Germany</span></h4>
        <h4 class="location"><img src="../assets/icons/phone.svg" alt="restaurant number" title="restaurant number"> <span>+49 40 57283299</span></h4>
      </div>
      <div class="retaurant-img"><img src="https://s3-media2.fl.yelpcdn.com/bphoto/RDKmsxZZczP4EiTCB9AYOw/o.jpg" alt="Billy The Butcher" title="Billy The Butcher"></div>
    </div>
    <div class="restaurant-reviews">
      <h5>Reviews</h5>
      <ul>
        <li>
          <p><b>Rating:</b> 4</p>
          <p>I've had decent lunches at Billy The Butcher a couple of times now. It's what I would describe as a trendy burger joint on the 4th floor of the Alsterhaus...</p>
        </li>
        <li>
          <p><b>Rating:</b> 4</p>
          <p>We arrived Hamburg at noon and decide to get some burgers before going to the office. This place is pretty close to the city hall and rated well.

            I ordered...</p>
        </li>
        <li>
          <p><b>Rating:</b> 4</p>
          <p>Nice place to have good burger for lunch. Most of personnel speak English which is good. Prices are decent, also I would recommend to share fries with a...</p>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`RestaurantDetail.vue RestaurantDetail home screen MatchSnapshot 1`] = `
<div>
  <div>
    <div class="restaurant-details">
      <div class="retaurant-header">
        <h2><a href="https://www.yelp.com/biz/billy-the-butcher-hamburg-2?adjust_creative=nMAD8ywU1QBl_RMSM215Bg&amp;utm_campaign=yelp_api_v3&amp;utm_medium=api_v3_graphql&amp;utm_source=nMAD8ywU1QBl_RMSM215Bg" target="_blank">Billy The Butcher</a></h2>
        <p><b>Rating:</b> 4</p>
        <h4 class="location"><img src="../assets/icons/marker.svg" alt="restaurant location" title="restaurant location"> <span>Jungfernstieg 16 - 20
4. OG
20354 Hamburg
Germany</span></h4>
        <h4 class="location"><img src="../assets/icons/phone.svg" alt="restaurant number" title="restaurant number"> <span>+49 40 57283299</span></h4>
      </div>
      <div class="retaurant-img"><img src="https://s3-media2.fl.yelpcdn.com/bphoto/RDKmsxZZczP4EiTCB9AYOw/o.jpg" alt="Billy The Butcher" title="Billy The Butcher"></div>
    </div>
    <div class="restaurant-reviews">
      <h5>Reviews</h5>
      <ul>
        <li>
          <p><b>Rating:</b> 4</p>
          <p>I've had decent lunches at Billy The Butcher a couple of times now. It's what I would describe as a trendy burger joint on the 4th floor of the Alsterhaus...</p>
        </li>
        <li>
          <p><b>Rating:</b> 4</p>
          <p>We arrived Hamburg at noon and decide to get some burgers before going to the office. This place is pretty close to the city hall and rated well.

            I ordered...</p>
        </li>
        <li>
          <p><b>Rating:</b> 4</p>
          <p>Nice place to have good burger for lunch. Most of personnel speak English which is good. Prices are decent, also I would recommend to share fries with a...</p>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
